### PR TITLE
add country and countrystate

### PR DESCRIPTION
**Pull request by MohammedMaher177**

### CODE REVIEW

File: YaqeenDAL/Model/Model.cs [hunk 1]
```diff
@@ -210,4 +210,29 @@ public class ResourceLocalization
         [Column(TypeName = "jsonb")]
         public Dictionary<string, string> Translation { get; set; }
     }
-}
\ No newline at end of file
+        public class Country
+    {
+        [Key]
```
Comment (EngHabu) [inline YaqeenDAL/Model/Model.cs:215]: Please add the Database generated annotation... It should be in other tables to copy from

File: YaqeenDAL/Model/Model.cs [hunk 2]
```diff
@@ -210,4 +210,29 @@ public class ResourceLocalization
...
+        public string AlphaCode { get; set; }
+    }
+    //country state
+           public class CountryState
+    {
+        [Key]
+        public int Id {get; set;}    
+        public string Name { get; set; }
+        public int CountryId { get; set; }
+        public string CountryCode { get; set; }
```
Comment (EngHabu) [inline YaqeenDAL/Model/Model.cs:227]: You don't need country code and country name here, right?

File: YaqeenDAL/Model/Model.cs [hunk 3]
```diff
@@ -210,4 +210,29 @@ public class ResourceLocalization
...
+        public string CountryName { get; set; }
+        public string StateCode { get; set; }
+        public string? Type { get; set; }
```
Comment (EngHabu) [inline YaqeenDAL/Model/Model.cs:230]: What's the type?

File: YaqeenDAL/Migrations/20231023120532_add-university.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+#nullable disable
+
+namespace YaqeenDAL.Migrations
+{
+    /// <inheritdoc />
+    public partial class adduniversity : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
```
Comment (EngHabu) [inline YaqeenDAL/Migrations/20231023120532_add-university.cs:11]: I think there is a problem in the workflow that generates migrations, these shouldn't be empty, can you generate them locally?
```
git pull
dotnet ef migrations remove -f
dotnet ef migrations add add-universities
git add -A
git commit -s -m "Generate migrations"
```

File: YaqeenDAL/Model/Model.cs (hunk 3 above)
Comment (enas48) [reply YaqeenDAL/Model/Model.cs]: type of country like  (province, territory, state, ...)

**Review by EngHabu — APPROVED** [review]

**Pull request merged**